### PR TITLE
[#93438736] Automate Ansible deployment using Jenkins

### PR DESCRIPTION
**Pull request by actionjack**
[Automate Ansible deployment using Jenkins](https://www.pivotaltracker.com/story/show/93438736)

**What**

When a change is made to [Ansible](https://github.com/ansible/ansible) playbook or a [Terraform](https://terraform.io/) run changes the state of the servers then [Jenkins](https://jenkins-ci.org/) should perform an `Ansible` run.

This PR request has the following components:
- Install the following required packages:
  - [Ansible](https://github.com/ansible/ansible) - our configuration management tool,
  - [Virtualenv](https://virtualenv.pypa.io) - to create isolated `Python` environments,
  - [Python](https://www.python.org/) - required by `ansible`,
  - [Figlet](http://www.figlet.org/) - to make all the things pretty.
- Disable [SSH strict host key checking](http://www.openbsd.org/cgi-bin/man.cgi/OpenBSD-current/man5/ssh_config.5?query=ssh_config&sec=5) for two reasons:
  1. We are creating and destroying environments that re-use the same hostnames and IP addresses,
  2. We do not prompt the `Jenkins` user to accept the host key of a newly created host.
- Setup [Google Compute Engine](https://cloud.google.com/compute/) [credentials for Ansible](http://docs.ansible.com/guide_gce.html#calling-modules-with-secrets-py)
- Create `Jenkins` jobs to run `Ansible` against:
  - [Amazon EC2](https://aws.amazon.com/)
  - [Google Compute Engine](https://cloud.google.com/compute/)
- Increase `Jenkins` [Vagrant Box](https://www.vagrantup.com/) memory to `448 Mb`
- Add instructions on:
  -  Setting up the `Google Compute Engine credentials for Ansible`
  -  Manually creating a `~/.vault_pass.txt` file for the `jenkins admin user`

**How to review this PR**

We've crafted this PR request to be reviewed in the following context:
- I need to install some packages in order to run ansible informatively, safely and securely using a non privileged `Jenkins` user,
- I need to stop `SSH` prompting me to accept changed host keys or approve keys from new hosts,
- I need to set the correct credentials for communicating with the [Google Compute Engine API](https://cloud.google.com/compute/docs/api/getting-started),
  - Credential support for [Amazon EC2](https://aws.amazon.com/) was already added in https://github.com/alphagov/multicloud-deploy/commit/dd14362a226a87b6382c5f90334e84b4429c2cd8 
- I need to create Jenkins jobs to run ansible against the [Continuous integration](http://en.wikipedia.org/wiki/Continuous_integration) environments on the following platforms:
  - `Amazon EC2`
  - `Google Compute Engine`
- I want to stop my Vagrant machine reporting ``virtual memory exhausted: Cannot allocate memory` when jenkins attempts to install python dependencies
- I want to know what I need to do in order to:
  - Put the correct variable properties in my `ansible vault file`
  - Manually to get `ansible` to decrypt the `ansible vault file`

**Who should review this PR**
- Anyone on the core team,
- I paired with @jimconner on this one so he has to keep his mitts off the merge button :stuck_out_tongue:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,8 +123,23 @@ keystore_password: "my.java.keystore.password"
...
+  GCE_KEYWORD_PARAMS = {'project': 'project_id'}  
+gce_account_certificate_pem |
+  -----BEGIN RSA PRIVATE KEY-----
+  -----END RSA PRIVATE KEY-----
 ```
 
+### Manual steps
+
+Your `jenkins_admin_user` will need to have a `~/.vault_pass.txt` file, the easiest way to create one is to do the following whilst logged in as the `jenkins_admin_user`:
+
```
Comment (mtekel) [inline README.md:138]: Did you actually mean `jenkins` user? This user is defined in the geerlingguy.jenkins ansible role. It's the default jenkins linux user. `jenkins_admin_user` in this readme refers to jenkins users that are also administrators.

File: README.md [hunk 2]
```diff
@@ -140,6 +155,8 @@ Some of the jobs we deploy require our AWS and [GCE](https://cloud.google.com/co
 * `aws_credentials` - AWS credentials in a form of shell include 
 * `gce_account` - google compute engine account.json file
 * `deployer_key:`- deployer key with elements: `name:` - name of the key and `key:` - the key itself
+* `secrets_py` - [Google Compute Engine](https://cloud.google.com/compute/) [credentials for Ansible](http://docs.ansible.com/guide_gce.html#calling-modules-with-secrets-py)
+* `gce_account_certificate_pem` - [Google Compute Engine](https://cloud.google.com/compute/)service account private key
```
Comment (mtekel) [inline README.md]: Looks like a missing space in front of 'service' (the link and 'service' are next to each other without space).

**Comment by actionjack** [comment]
@mtekel Fixed the issues in the README.md file.

**Comment by mtekel** [comment]
I think we also want to enable mail notifications for both jobs. This can be done by adding: 

```
  publishers {
    mailer("the-multi-cloud-paas-team@digital.cabinet-office.gov.uk", false, true)
  }
```

**Comment by actionjack** [comment]
@mtekel mailers added.

**Comment by dcarley** [comment]
#12 caused this to need a rebase, sorry!

**Comment by mtekel** [comment]
After https://github.com/alphagov/tsuru-ansible/pull/57 gce job will need update:

**ANSIBLE_USE_PRIVATE_SSH=true** SSL_CERT_FILE=$(python -m certifi) ansible-playbook   -i gce.py   --vault-password-file  ~/.vault_pass.txt   -e "deploy_env=${DEPLOY_ENV}"   -e "@platform-gce.yml"   site-gce.yml

**Comment by mtekel** [comment]
Just tested all again and it works fine, merging.

**Pull request merged**